src/components/RegistryInfoContent/RegistryInfoContent.tsx

```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
 import { CommandContainer } from './styles';
 import { Props, State } from './types';
 
-/* eslint react/prop-types:0 */
-function TabContainer({ children }): JSX.Element {
-  return (
-    <CommandContainer>
-      <Typography
-        className={css`
-          padding: 0;
-          min-height: 170;
-        `}
-        component="div">
-        {children}
-      </Typography>
-    </CommandContainer>
-  );
-}
-
-class RegistryInfoContent extends Component<Props, State> {
-  public state = {
-    tabPosition: 0,
-  };
-
-  public render(): JSX.Element {
-    return <div>{this.renderTabs()}</div>;
-  }
-
-  private handleChange = (event: React.ChangeEvent<{}>, tabPosition: number) => {
+export const RegistryInfoContent: React.FC<Props> = props => {
+  const [tabPosition, setTabPosition] = useState<State['tabPosition']>(0);
```

Comment:
```suggestion
  const [tabPosition, setTabPosition] = useState(0);
```

Here you don't need to set a type, it will be automatically of type number, because of it's value, 0.

Reply:
Nothing wrong with being explicit.
Setting a type is useful so that you can prevent a typo like this:
```typescript
const [tabPosition, setTabPosition] = useState('0'); //ok...
```
Without explicitly setting a type the compiler is fine with whatever you initialize the state with.
Now the type is `string` and any error because of type mismatch will show up where you consume the state, whilst if you would have been explicit:
```typescript
const [tabPosition, setTabPosition] = useState<number>('0'); //Argument of type '"0"' is not assignable to parameter of type 'number | (() => number)'
```
You would get an error right on this line, where the actual error happenend, since you **intended** the state to be of type `number` but (accidentally) passed in a `string` instead. :wink:

Reply:
I agree although typescript is smart enough to infer the type here we should declare it. if not then we should remove our state interface.

Reply:
Hmmm I find it unnecessary, because we are not really setting this state so dynamically.... and even if we were passing a prop, it would be checked though of the interface or if we pass for example a string in the setTabPosition we will get a type error... so IMO is fine if you want to cast the state, but  it is very verbose and unnecessary.

Reply:
understood